File: internal/templates/templates/webhook-relay/_node/services/receive-webhook/handler.mjs
// POST /webhooks — the webhook golden rule: ack fast, process later.
//
// Stripe/GitHub/etc. expect a quick 2xx; anything slow or flaky belongs in
// the queue, where retries are free.
import { randomUUID } from "node:crypto";
import { SQSClient, GetQueueUrlCommand, SendMessageCommand } from "@aws-sdk/client-sqs";

const sqs = new SQSClient({});
const { QueueUrl } = await sqs.send(new GetQueueUrlCommand({ QueueName: process.env.QUEUE_NAME }));

export const handler = async (event) => {
  const payload = JSON.parse(event.body ?? "{}");
  const delivery = { id: randomUUID(), ...payload };

  await sqs.send(new SendMessageCommand({ QueueUrl, MessageBody: JSON.stringify(delivery) }));

  console.log(`webhook ${delivery.id} queued (${payload.type ?? "unknown"})`);
  return { statusCode: 202, body: JSON.stringify({ received: delivery.id }) };
};

File: internal/templates/templates/webhook-relay/_python/services/receive-webhook/handler.py
"""POST /webhooks — the webhook golden rule: ack fast, process later.

Stripe/GitHub/etc. expect a quick 2xx; anything slow or flaky belongs in
the queue, where retries are free.
"""

import json
import os
import uuid

import boto3

sqs = boto3.client("sqs")
queue_url = sqs.get_queue_url(QueueName=os.environ["QUEUE_NAME"])["QueueUrl"]


def handler(event, context):
    payload = json.loads(event.get("body") or "{}")
    delivery = {"id": str(uuid.uuid4()), **payload}

    sqs.send_message(QueueUrl=queue_url, MessageBody=json.dumps(delivery))

    print(f"webhook {delivery['id']} queued ({payload.get('type', 'unknown')})")
    return {"statusCode": 202, "body": json.dumps({"received": delivery["id"]})}
